src/app/pages/questions/components/question/audio-input/audio-input.component.ts

```diff
@@ -42,43 +39,54 @@ export class AudioInputComponent implements OnInit, OnChanges {
   value: string = null
   configFile = 'liveinput_android.conf'
   compression = 1
-  platform = false
   answer_b64: string = null
   permission: any
   avgWordsPerMinute = 100
-
-  ngOnInit() {}
-
-  ngOnChanges() {
-    if (this.currentlyShown) {
-      this.startRecording()
-    }
-  }
+  alertShown = false
+  resumeListener
+  pauseListener
+  recordingTimeout
 
   constructor(
-    public questions: QuestionsPageComponent,
     private audioRecordService: AudioRecordService,
     private permissionUtil: AndroidPermissionUtility,
     public navCtrl: NavController,
     public alertCtrl: AlertController,
-    private device: Device
+    private platform: Platform,
+    private translate: TranslatePipe
   ) {
-    // NOTE: Stop audio recording when application is on pause / backbutton is pressed
-    document.addEventListener('pause', () => {
-      console.log('on pause')
-      // if (this.navCtrl.isActive(this.navCtrl.getActive()) == false) {
-      this.audioRecordService.stopAudioRecording()
-      // }
-    })
+    this.permissionUtil.checkPermissions()
+  }
 
-    document.addEventListener('backbutton', () => {
-      console.log('on backbutton')
-      // if (this.navCtrl.isActive(this.navCtrl.getActive()) == false) {
-      this.audioRecordService.stopAudioRecording()
-      // }
-    })
+  ngOnInit() {
+    this.onResume()
+    this.onClose()
+  }
 
-    this.permissionUtil.checkPermissions()
+  ngOnChanges() {
+    if (this.currentlyShown) this.startRecording()
+  }
+
+  ngOnDestroy() {
+    this.resumeListener.unsubscribe()
+    this.pauseListener.unsubscribe()
+  }
+
+  onResume() {
```

Comment:
It may be better to put this directly in the constructor or change the name of the method. As the `onResume()` name is confusing ( like this is automatically fired by ionic when resumed)

Reply:
Okay, I have directly moved it to ngOnInit.